CMakeModules/Findpypy.cmake

```diff
@@ -5,8 +5,18 @@ include(FindPackageHandleStandardArgs)
 find_program(PYPY NAMES pypy${PYTHON_VERSION_MAJOR} pypy)
 
 if(PYPY)
+  execute_process(
+    COMMAND ${PYPY} "-c" "import platform; print(platform.python_version())"
+    OUTPUT_VARIABLE PYPY_PYTHON_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  message(WARNING ${PYPY_PYTHON_VERSION})
+  if(${PYPY_PYTHON_VERSION} VERSION_GREATER_EQUAL ${PYTHON_VERSION_FULL})
```

Comment:
Is this necessary? I typically install python and pypy via a package manager. Pypy's version is typically older than the python version and this hasn't caused issues for me.

Reply:
When we were developing a `conda` target we ran into issues with respect to Python versions in the `conda` environment, and `pypy` in the base environment leaking through `conda`'s path (I believe with Python 3.10 in the `conda` env). @wbthomason can elaborate more.

Reply:
Yes - `conda` seems to be a leakier abstraction than we hoped, and having `pypy` for Python 2 installed through the system package manager while trying to build the bindings for Python 3 caused failures that motivated this check. I forget why we're testing against `PYTHON_VERSION_FULL` and not just `PYTHON_VERSION_MAJOR`, though...perhaps because there's no equivalent for `pypy`? I can experiment and check if this can be relaxed.

@mamoll, when you say that your `pypy` version is usually older than the Python version, do you mean different minor versions (e.g. `pypy` at 3.9, Python at 3.10) or major versions?

Reply:
Actually, I think @mamoll is right here - we already restrict to `pypy3` on line 4. I've just run an experiment building the bindings with and without `pypy`, with this conditional removed, inside `conda`, and it seems to have worked.

@zkingston if you can confirm, I think we can remove this check.

Reply:
@mamoll @wbthomason I changed the check to instead just check the major version - there is an issue if pypy does not match (which can happen if pypy2 is installed with a Python 3 executable).

Reply:
@wbthomason yes, I think major version match between pypy and python is sufficient (at least, that has been my experience).